File: scripts/python/boot.py
# -*- coding: utf-8 -*-
import numpy as np
import pandas as pd
import pyper as pr

# Leitura dos dados
matriz = pd.read_csv("especies.csv", header=0, index_col=0, encoding="utf8")

# Transformação logarítmica
matriz_log = matriz.apply(lambda x: np.log10(x + 1)) # Logaritmos decimais

# Teste de significância dos agrupamentos
r = pr.R(RCMD='C:\\Program Files\\R\\R-3.5.2\\bin\\R', use_pandas=True)
r.assign('rdata', matriz_log)
r('library(pvclust)')
r('result <- pvclust(rdata, method.dist="euclidean", method.hclust="average", nboot=1000)')
r('png("boot.png"); plot(result); dev.off()')
r('pvrect(result, alpha=0.95)')
print(r('result'))

File: scripts/r/boot.R
# Leitura dos dados
dados <- read.csv("especies.csv", row.names=1) #header = TRUE)

# Construção da matriz de dados
matriz <- as.data.frame(dados)

# Transposição da matriz de dados
#matriz <- t(matriz)

# Transformação logarítmica
matriz.log <- log10(matriz + 1)

# Cálculo da distância de Bray-Curtis entre as amostras
#library(vegan)
#dist <- vegdist(matriz.log, method="bray", binary=FALSE, diag=TRUE)

# Teste de significância dos agrupamentos
#library(fpc)
#set.seed(20000)
#cf <- clusterboot(dist, 
#                  B=100, 
#                  distances=TRUE, 
#                  bootmethod="boot",
#                  clustermethod=disthclustCBI,
#                  k=2, cut="number", method="average",
#                  showplots=FALSE,
#                  seed=15555)
#print(cf)
#plot(cf)

library(pvclust)
result <- pvclust(matriz.log, method.dist="euclidean", method.hclust="average", nboot=1000)
plot(result)
pvrect(result, alpha=0.95)